server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -321,6 +328,64 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         this.cachedTimeThread.start();
     }
 
+    private static HashMap<String, LongGauge> setupMetrics(MeterRegistry meterRegistry, String name, ExecutorHolder holder) {
+        Map<String, Object> at = null;
+        HashMap<String, LongGauge> gauges = new HashMap<>();
```

Comment:
It's not clear to me why this is a map - we are never using the keys, only the values. Maybe a List will be enough?

Reply:
Changed to an ArrayList